### ISSUE TITLE
hackEarlySend broken on FabricAPI 0.93+

### ISSUE THREAD

**Issue by ItzSkyReed** [OKTW-Network/FabricProxy-Lite#77]
got this error "Unexpected custom data from client" on 1.20.4, trying different version of velocity 3.3.0: build 311, 320, 316
 
 my config:
 ```
 hackOnlineMode = true
hackEarlySend = true
hackMessageChain = true
disconnectMessage = "This server requires you to connect with Velocity."
```

**Comment by ItzSkyReed** [comment]
problem was in hackEarlySend, after I set it to false everything start working fine

**Comment by Strongis13** [comment]
> problem was in hackEarlySend, after I set it to false everything start working fine

I am having this issue too, but disabling hackEarlySend means I cannot use LuckPerms. Are there any other workarounds to the "Unexpected custom data from client" error?

**Comment by james58899** [comment]
This is a bug in FabricAPI. It can be workaround by downgrade FabricAPI to 0.92.1

**Comment by james58899** [comment]
Fixed in FabricAPI 0.94.1
